django/urls/resolvers.py

```diff
@@ -90,6 +90,11 @@ def __init__(self, regex):
         # expression.
         self._regex = regex
         self._regex_dict = {}
+        # As a performance optimization, if the given regex string is a regular
+        # string (not a lazily-translated string proxy), we can just compile it
+        # once and avoid the property and the per-langage compilation entirely.
+        if isinstance(regex, six.string_types):
+            self.__dict__['regex'] = self._compile(regex)
```

Comment:
It doesn't work with `property` as it defines a `__set__` method. Using a readonly descriptor (only defining `__get__` à la `cached_property`) should work here and be worth it as it would prevent a function call in what seems to a be a performance critical path.